### PR TITLE
[Arm64] Add Simd HWIntrinsic test

### PR DESCRIPTION
**Pull request by sdmaclea**

### CODE REVIEW

**Comment by sdmaclea** [comment]
@tannergooding I was wondering about you opinion about this versus #15771.  In some ways it is similar, but perhaps it is a slightly different problem.  

I am not sure what advantage a script might offer or whether a common script could be used to generate tests for all platforms.

File: tests/src/JIT/HardwareIntrinsics/Arm64/Simd.cs [hunk 1]
```diff
@@ -222,8 +222,13 @@ class Program
             {
                 binOp(v,v);
             }
-            catch (PlatformNotSupportedException) // TODO-FIXME
+            catch (NotSupportedException e)
             {
+                if (!e.Message.Equals("Type is not supported."))
```
Comment (tannergooding) [inline tests/src/JIT/HardwareIntrinsics/Arm64/Simd.cs]: This doesn't take localization into account, not sure if that's an issue.

Comment (sdmaclea) [reply tests/src/JIT/HardwareIntrinsics/Arm64/Simd.cs]: I was aware of that.  For initial testing it was sufficient to prove the message was set correctly.  

I'll drop it after #16102 merges.

**Comment by tannergooding** [comment]
> I am not sure what advantage a script might offer or whether a common script could be used to generate tests for all platforms.

The benefit of using a script/templates for generation is that it lets us easily add new scenarios "en masse".

That is, many of the intrinsics take the same number of inputs and the only thing that differs is:
* The method you call
* The operation you do when validating the result

For example, `Sse.Add`, `Sse.Multiply`, `Sse.Subtract`, and `Sse.Divide` only differ in the validation by `+`, `*`, `-`, or `/`. Otherwise, they undergo all the same validation scenarios, etc.

The script and template means that I write the code once, run the script, and get accurate tests for all 4. I can then add any new scenarios to the template, run the script again, and have all 4 tests updated (and this actually scales much farther than just 4).

**Comment by tannergooding** [comment]
It should definitely be possible to share the generation scripts/templates between the architectures.

**Comment by sdmaclea** [comment]
@tannergooding Would it be useful to Merge these tests?

I could move the test body inside a 
```C#
#if ARM64_SIMD_API_PENDING_APPROVAL_AND_COREFX_MERGE
#endif
```

**Comment by sdmaclea** [comment]
I marked the individual tests with 
`ARM64_SIMD_API_PENDING_APPROVAL_AND_OR_COREFX_MERGE`

@CarolEidt @tannergooding 
I think we should review and merge this.  
+ It should pass on tip (since it doesn't do anything).
+ It passes on Arm64/Linux with modified CoreFX & `ARM64_SIMD_API_PENDING_APPROVAL_AND_OR_COREFX_MERGE` defined
+ It will allow the tests to be generated by scripts like #15771.
+ It will allow future tests to be reviewed as intrinsics are implemented in CoreCLR.

**Comment by tannergooding** [comment]
@sdmaclea, I'll give a review pass sometime later today when I get some time.

I will defer to @CarolEidt on whether we should merge these before or after API review/approval

**Comment by sdmaclea** [comment]
@CarolEidt Can you comment on the strategy of reviewing and merging these tests while they are disabled.  It would allow the tests to be in the master branch to enable some collaboration.

**Comment by CarolEidt** [comment]
> Can you comment on the strategy of reviewing and merging these tests while they are disabled. It would allow the tests to be in the master branch to enable some collaboration.

I think it's a reasonable thing to do.

**Comment by CarolEidt** [comment]
I've taken a look at this, and it seems fine to me. I didn't scrutinize it deeply, as I hope that we will replace this with a shared template between architectures.

**Comment by sdmaclea** [comment]
@CarolEidt lets merge this now that it is passing.

**Pull request merged**